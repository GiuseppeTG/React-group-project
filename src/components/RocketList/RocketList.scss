* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.rocket-list-container {
  margin-top: 4rem;
  padding: 4rem;
}

.rocket-list {
  display: flex;
  flex-direction: column;
  gap: 2rem;
}