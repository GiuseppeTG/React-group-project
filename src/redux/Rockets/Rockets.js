const rocketsAPI = 'https://api.spacexdata.com/v3/rockets/';

// Actions

const GET_ROCKETS = 'redux/Rockets/Rockets/GET_ROCKETS';

// Reducer

export default function rocketsReducer(state = [], action = {}) {
  switch (action.type) {
    case GET_ROCKETS:
      return action.payload;
    default:
      return state;
  }
}

// Action Creator

function getRockets() {
  return async (dispatch) => {
    const response = await fetch(rocketsAPI);
    const data = await response.json();
    const rockets = [];
    data.forEach((rocket) => {
      const newRocket = {
        id: rocket.rocket_id,
        name: rocket.rocket_name,
        description: rocket.description,
        img: rocket.flickr_images[0],
        reserved: false,
      };
      rockets.push(newRocket);
    });
    dispatch({
      type: GET_ROCKETS,
      payload: rockets,
    });
  };
}

export { getRockets };
