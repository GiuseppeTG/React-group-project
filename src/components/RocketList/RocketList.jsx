import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './RocketList.scss';
import Rocket from '../Rocket/Rocket';
import { getRockets } from '../../redux/Rockets/Rockets';

export default function RocketList() {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRockets());
  }, []);

  return (
    <div className="rocket-list-container">
      <ul className="rocket-list">
        {rockets.map((rocket) => (
          <Rocket
            key={rocket.id}
            rocket={rocket}
          />
        ))}
      </ul>
    </div>
  );
}
