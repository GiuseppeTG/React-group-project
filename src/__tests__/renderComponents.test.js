import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import RocketList from '../components/RocketList/RocketList';
import Rockets from '../Pages/Rockets';
import store from '../redux/configureStore';

describe('Testing component rednering propperly', () => {
  test('renders Rockets component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Rockets />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('renders RocketList component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <RocketList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
