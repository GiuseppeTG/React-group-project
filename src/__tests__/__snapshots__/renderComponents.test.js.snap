// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component rednering propperly renders MissionsList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        MissionList
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      MissionList
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders RocketList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket-list-container"
        data-testid="rocket-list-test"
      >
        <h2
          class="list-title"
        >
          ROCKETS
        </h2>
        <ul
          class="rocket-list"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket-list-container"
      data-testid="rocket-list-test"
    >
      <h2
        class="list-title"
      >
        ROCKETS
      </h2>
      <ul
        class="rocket-list"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders Rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket-list-container"
        data-testid="rocket-list-test"
      >
        <h2
          class="list-title"
        >
          ROCKETS
        </h2>
        <ul
          class="rocket-list"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket-list-container"
      data-testid="rocket-list-test"
    >
      <h2
        class="list-title"
      >
        ROCKETS
      </h2>
      <ul
        class="rocket-list"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <div
          class="logo-and-text"
        >
          <img
            alt="planet logo"
            src="planetLogo.png"
          />
          <p>
            <a
              aria-current="page"
              class="logo-text active"
              href="/"
            >
              Space Travelers' Hub
            </a>
          </p>
        </div>
        <ul
          class="links-list"
        >
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              ROCKETS
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/missions"
            >
              MISSIONS
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/profile"
              id="my-profile"
            >
              MY PROFILE
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <div
        class="logo-and-text"
      >
        <img
          alt="planet logo"
          src="planetLogo.png"
        />
        <p>
          <a
            aria-current="page"
            class="logo-text active"
            href="/"
          >
            Space Travelers' Hub
          </a>
        </p>
      </div>
      <ul
        class="links-list"
      >
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            ROCKETS
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/missions"
          >
            MISSIONS
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/profile"
            id="my-profile"
          >
            MY PROFILE
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
